### DISCUSSION TITLE
Is it possible to use camelcase in form objects?

### DISCUSSION THREAD

**Discussion by michael-lajos** [livewire/livewire#8011] [1. Help] [answered]
I am using form objects in Livewire v3. My `Project.php` model has a property called `time_justification`. In the form object I am using camelcase for my property names. 

```php
class ProjectForm extends Form
{
    #[Validate('required|string|max:255')]
    public $name = '';

    #[Validate('required|boolean')]
    public $timeJustification = false;
}
```

However when calling 

```php
Project::create(
    $this->form->all()
);
```

I get the error: **Field 'time_justification' doesn't have a default value**

I guess this is because behind the screens camelCase is not converted to snake_case?

Is there an easy way to map this or should just use snake_case in my **Livewire formobject**?

Any advice is appreciated! ✌️

**Comment by joshhanley** [comment, marked as answer]
@michael-lajos the problem is `->all()` returns an array of the public properties on the form object, with the property name as a key and the value being the properties value.

If your properties are camelCased `$myProperty`, then the array will be `['myProperty' => 'someValue']`, but they are snaked_cased `$my_property`, then the array will be `['my_property' => 'someValue']`.

Using mass assigning like `Post::create()` requires the array keys to be exactly the same as the attributes on the model, which is snake_cased.

So you will either need to use `$snaked_cased` properties on your form object, or convert the keys to snake_case before you pass it to the created method. You could do that by overriding the `->all()` method in your form object and doing the translation there.

Hope this helps!

**Reply by michael-lajos** [reply]
@joshhanley thanks for your detailed answer! In the end I opted for the `$snaked_cased` properties on the form object. It does leave me a bit conflicted to use snake_case property names. 

I wonder if it would be a good idea to implement conversion under the hood in Laravel. Kinda like an inverse of how Laravel accessors used to work. 

```php
 public function getMyCustomAttribute()
    {
        return 'custom value';  
    }
```

Which was accessible by calling `$model->my_custom`;

**Reply by joshhanley** [reply]
@michael-lajos no worries! 🙂 Yeah I know plenty of people now using snake_cased properties. Still makes me feel weird too, but makes sense for things that work with Laravel models. And if you're doing some classes like that, then why not be consistent across the codebase... but yeah it's weird 😆
